providers/nitro/nitro-helper/src/shared.rs

```diff
@@ -28,6 +28,32 @@ pub struct NitroConfig {
     pub aws_region: String,
 }
 
+#[derive(Debug, Serialize, Deserialize)]
+pub struct NitroKeygenConfig {
+    /// AWS credentials -- if not set, they'll be obtained from IAM
+    pub credentials: AwsCredentials,
+    /// AWS key id
+    pub kms_key_id: String,
+    /// AWS region
+    pub aws_region: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct NitroKeygenResponse {
+    /// the encrypted private key
+    pub cipher_privkey: Vec<u8>,
+    /// the public key
+    pub pubkey: Vec<u8>,
+    /// error info
+    pub error: String,
```

Comment:
I assume the response will either be `cipher_privkey` and `pubkey` are entered, or `error` is returned?
so it could be an enum or perhaps a type alias to `Result<(Vec<u8>, Vec<8>), String>` ?